sdk/core/azure-core/test/ut/logging.cpp

```diff
@@ -1,112 +1,225 @@
-  }
+      Log(LogLevel::Verbose, "Verbose");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "");
 
-  EXPECT_EQ(logRecorder.Actual, expected);
-}
+      Log(LogLevel::Informational, "Informational");
+      EXPECT_EQ(level, LogLevel::Informational);
+      EXPECT_EQ(message, "Informational");
 
-TEST(Logging, noClassifications)
-{
-  LogArguments const Values{
-      {Http::LogClassification::Request, "Request"},
-      {Http::LogClassification::Request, "Request"},
-      {Http::LogClassification::Response, "Response"},
-      {Http::LogClassification::Request, "Request"},
-  };
+      Log(LogLevel::Warning, "Warning");
+      EXPECT_EQ(level, LogLevel::Warning);
+      EXPECT_EQ(message, "Warning");
+
+      Log(LogLevel::Error, "Error");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "Error");
+    }
+
+    SetLogLevel(LogLevel::Warning);
+    {
+      level = LogLevel::Error;
+      message = "";
+
+      Log(LogLevel::Verbose, "Verbose");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "");
+
+      Log(LogLevel::Informational, "Informational");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "");
 
-  LogRecorder logRecorder;
-  Logging::SetLogListener(logRecorder.LogListener);
+      Log(LogLevel::Warning, "Warning");
+      EXPECT_EQ(level, LogLevel::Warning);
+      EXPECT_EQ(message, "Warning");
 
-  auto const AllowedClassification = Http::LogClassification::Request;
-  Logging::SetLogClassifications(Logging::LogClassification::None);
-  LogArguments const expected; // Empty
+      Log(LogLevel::Error, "Error");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "Error");
+    }
+
+    SetLogLevel(LogLevel::Error);
+    {
+      level = LogLevel::Error;
+      message = "";
+
+      Log(LogLevel::Verbose, "Verbose");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "");
+
+      Log(LogLevel::Informational, "Informational");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "");
+
+      Log(LogLevel::Warning, "Warning");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "");
+
+      level = LogLevel::Verbose;
+
+      Log(LogLevel::Error, "Error");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "Error");
+    }
+
+    SetLogLevel(LogLevel::Verbose);
+    {
+      level = LogLevel::Error;
+      message = "";
+
+      Log(LogLevel::Verbose, "Verbose");
+      EXPECT_EQ(level, LogLevel::Verbose);
+      EXPECT_EQ(message, "Verbose");
+
+      Log(LogLevel::Informational, "Informational");
+      EXPECT_EQ(level, LogLevel::Informational);
+      EXPECT_EQ(message, "Informational");
 
-  for (auto value : Values)
+      Log(LogLevel::Warning, "Warning");
+      EXPECT_EQ(level, LogLevel::Warning);
+      EXPECT_EQ(message, "Warning");
+
+      Log(LogLevel::Error, "Error");
+      EXPECT_EQ(level, LogLevel::Error);
+      EXPECT_EQ(message, "Error");
+    }
```

Comment:
This block is duplicated above, remove?
We have four levels, so I expected four such blocks and we have five.

Reply:
No, we verify that you can set it back to verbose. Maybe it does not make as much sense, but that was intentional.